src/fly/sorts.rs

```diff
@@ -19,100 +20,100 @@ pub enum SortError {
 
     #[error("expected {0} but found {1}")]
     NotEqual(Sort, Sort),
+    #[error("expected {0} args but found {1} args")]
+    ArgMismatch(usize, usize),
 
-    #[error("expected {1} args but found {2} args for {0}")]
-    ArgMismatch(String, usize, usize),
     #[error("{0} expected args but didn't get them")]
     Uncalled(String),
     #[error("{0} was called but didn't take any args")]
     Uncallable(String),
 
-    #[error("sort inference isn't supported yet")]
-    NoInference,
-}
-
-// helper function to clean up repeated if statements
-fn sort_eq(a: &Sort, b: &Sort) -> Result<(), SortError> {
-    if a == b {
-        Ok(())
-    } else {
-        Err(SortError::NotEqual(a.clone(), b.clone()))
-    }
+    #[error("function sort inference is not currrently supported")]
+    UnknownFunctionSort,
+    #[error("could not solve for the sort of {0}")]
+    UnsolvedSort(String),
 }
 
 // entry point for the sort checker
```

Comment:
You should probably add a doc string here, especially to explain the modifications made to `module`.

I also think the name should capture that this is sort checking and inference. Perhaps `infer` would actually be a better name.